daphne/src/roles_test.rs

```diff
@@ -361,92 +526,106 @@ async fn http_get_hpke_config_missing_task_id() {
     let mut rng = thread_rng();
-    let mut req = helper.gen_test_agg_cont_req(Id(rng.gen()), Vec::default());
+    let mut req = t.gen_test_agg_cont_req(Id(rng.gen()), Vec::default()).await;
+    req.sender_auth = None;
 
     // Expect failure due to missing bearer token.
     assert_matches!(
-        helper.http_post_aggregate(&req).await,
+        t.helper.http_post_aggregate(&req).await,
         Err(DapAbort::UnauthorizedRequest)
     );
 
     // Expect failure due to incorrect bearer token.
     req.sender_auth = Some(BearerToken::from("incorrect auth token!".to_string()));
     assert_matches!(
-        helper.http_post_aggregate(&req).await,
+        t.helper.http_post_aggregate(&req).await,
         Err(DapAbort::UnauthorizedRequest)
     );
 }
 
 #[tokio::test]
 async fn http_post_aggregate_share_unauthorized_request() {
-    let helper = MockAggregator::new();
-    let mut req = helper.gen_test_agg_share_req(0, [0; 32]);
+    let t = Test::new();
+    let mut req = t.gen_test_agg_share_req(0, [0; 32]).await;
+    req.sender_auth = None;
 
     // Expect failure due to missing bearer token.
     assert_matches!(
-        helper.http_post_aggregate_share(&req).await,
+        t.helper.http_post_aggregate_share(&req).await,
         Err(DapAbort::UnauthorizedRequest)
     );
 
     // Expect failure due to incorrect bearer token.
     req.sender_auth = Some(BearerToken::from("incorrect auth token!".to_string()));
     assert_matches!(
-        helper.http_post_aggregate_share(&req).await,
+        t.helper.http_post_aggregate_share(&req).await,
         Err(DapAbort::UnauthorizedRequest)
     );
 }
 
 #[tokio::test]
 async fn http_post_collect_unauthorized_request() {
-    let leader = MockAggregator::new();
-    let mut req = leader.gen_test_collect_req();
+    let t = Test::new();
+    let task_id = &t.time_interval_task_id;
+    let task_config = t.leader.tasks.get(task_id).unwrap();
+    let mut req = DapRequest {
+        version: task_config.version,
+        media_type: Some(MEDIA_TYPE_COLLECT_REQ),
+        payload: CollectReq {
+            task_id: task_id.clone(),
+            query: Query::default(),
+            agg_param: Vec::default(),
+        }
+        .get_encoded(),
+        url: task_config.leader_url.join("collect").unwrap(),
+        sender_auth: None, // Unauthorized request.
+    };
 
     // Expect failure due to missing bearer token.
     assert_matches!(
-        leader.http_post_collect(&req).await,
+        t.leader.http_post_collect(&req).await,
         Err(DapAbort::UnauthorizedRequest)
     );
 
     // Expect failure due to incorrect bearer token.
     req.sender_auth = Some(BearerToken::from("incorrect auth token!".to_string()));
     assert_matches!(
-        leader.http_post_collect(&req).await,
+        t.leader.http_post_collect(&req).await,
         Err(DapAbort::UnauthorizedRequest)
     );
 }
 
 #[tokio::test]
 async fn http_post_aggregate_failure_hpke_decrypt_error() {
```

Comment:
Unrelated: do we have a test for sending reports with an unknown config ID?

Reply:
Nope, and in fact we weren't enforcing this! I added tests to daphne/src/vdaf/mod_test.rs and added the check to the implementation of HpkeDecrypter for HpkeReceiverConfig.

Reply:
Good catch.